pkg/cfn/builder/managed_nodegroup_test.go

```diff
@@ -77,12 +78,14 @@ func TestManagedPolicyResources(t *testing.T) {
 			clusterConfig := api.NewClusterConfig()
 
 			ng := api.NewManagedNodeGroup()
+			api.SetManagedNodeGroupDefaults(ng, clusterConfig.Metadata)
 			ng.IAM.WithAddonPolicies = tt.addons
 			ng.IAM.AttachPolicyARNs = prefixPolicies(tt.attachPolicyARNs...)
 
 			p := mockprovider.NewMockProvider()
 			fakeVPCImporter := new(vpcfakes.FakeImporter)
-			stack := NewManagedNodeGroup(p.EC2(), clusterConfig, ng, nil, false, fakeVPCImporter)
+			bootstrapper := nodebootstrap.NewManagedBootstrapper(clusterConfig, ng)
```

Comment:
same here these should probably be fakes

Reply:
A fake doesn't achieve much here and using a real bootstrapper also ensures the generated template more closely resembles a working nodegroup template.

Reply:
the bootstrapper interface exists to decouple it from the builder, so if i change something in the bootstrapper it doesn;t break things in the builder